### PR TITLE
Add GET/{manager,cluster/:node_id}/config/:component/:configuration endpoints

### PR DESCRIPTION
**Pull request by mgmacias95**
Hello team,

This PR closes https://github.com/wazuh/wazuh/issues/2988.

Unit tests:
```python
# /var/ossec/framework/python/bin/pytest tests/test_agent.py -vv
============================================================================================================================ test session starts ============================================================================================================================
platform linux -- Python 3.7.2, pytest-4.4.0, py-1.8.0, pluggy-0.9.0 -- /var/ossec/framework/python/bin/python3.7
cachedir: .pytest_cache
rootdir: /home/vagrant/wazuh_api/wazuh/framework
collected 35 items

tests/test_agent.py::test_get_agents_overview_default PASSED                                                                                                                                                                                                          [  2%]
tests/test_agent.py::test_get_agents_overview_select[select0-all-None-0] PASSED                                                                                                                                                                                       [  5%]
tests/test_agent.py::test_get_agents_overview_select[select1-all-None-1] PASSED                                                                                                                                                                                       [  8%]
tests/test_agent.py::test_get_agents_overview_select[select2-all-None-1] PASSED                                                                                                                                                                                       [ 11%]
tests/test_agent.py::test_get_agents_overview_select[select3-Active,Pending-None-0] PASSED                                                                                                                                                                            [ 14%]
tests/test_agent.py::test_get_agents_overview_select[select4-Disconnected-None-1] PASSED                                                                                                                                                                              [ 17%]
tests/test_agent.py::test_get_agents_overview_select[select5-Disconnected-1s-1] PASSED                                                                                                                                                                                [ 20%]
tests/test_agent.py::test_get_agents_overview_select[select6-Disconnected-2h-0] PASSED                                                                                                                                                                                [ 22%]
tests/test_agent.py::test_get_agents_overview_select[select7-all-15m-2] PASSED                                                                                                                                                                                        [ 25%]
tests/test_agent.py::test_get_agents_overview_select[select8-Active-15m-0] PASSED                                                                                                                                                                                     [ 28%]
tests/test_agent.py::test_get_agents_overview_select[select9-Active,Pending-15m-1] PASSED                                                                                                                                                                             [ 31%]
tests/test_agent.py::test_get_agents_overview_select[select10-status10-15m-1] PASSED                                                                                                                                                                                  [ 34%]
tests/test_agent.py::test_get_agents_overview_query[ip=172.17.0.201] PASSED                                                                                                                                                                                           [ 37%]
tests/test_agent.py::test_get_agents_overview_query[ip=172.17.0.202] PASSED                                                                                                                                                                                           [ 40%]
tests/test_agent.py::test_get_agents_overview_query[ip=172.17.0.202;registerIP=any] PASSED                                                                                                                                                                            [ 42%]
tests/test_agent.py::test_get_agents_overview_query[status=Disconnected;lastKeepAlive>34m] PASSED                                                                                                                                                                     [ 45%]
tests/test_agent.py::test_get_agents_overview_query[(status=Active,status=Pending);lastKeepAlive>5m] PASSED                                                                                                                                                           [ 48%]
tests/test_agent.py::test_get_agents_overview_search[search0-3] PASSED                                                                                                                                                                                                [ 51%]
tests/test_agent.py::test_get_agents_overview_search[search1-3] PASSED                                                                                                                                                                                                [ 54%]
tests/test_agent.py::test_get_agents_overview_search[search2-1] PASSED                                                                                                                                                                                                [ 57%]
tests/test_agent.py::test_get_agents_overview_search[search3-5] PASSED                                                                                                                                                                                                [ 60%]
tests/test_agent.py::test_get_agents_overview_search[search4-2] PASSED                                                                                                                                                                                                [ 62%]
tests/test_agent.py::test_get_agents_overview_status_olderthan[active-9m-1-None] PASSED                                                                                                                                                                               [ 65%]
tests/test_agent.py::test_get_agents_overview_status_olderthan[all-1s-5-None] PASSED                                                                                                                                                                                  [ 68%]
tests/test_agent.py::test_get_agents_overview_status_olderthan[pending,neverconnected-30m-1-None] PASSED                                                                                                                                                              [ 71%]
tests/test_agent.py::test_get_agents_overview_status_olderthan[55-30m-0-1729] PASSED                                                                                                                                                                                  [ 74%]
tests/test_agent.py::test_get_config_error[100-logcollector-internal-1701] PASSED                                                                                                                                                                                     [ 77%]
tests/test_agent.py::test_get_config_error[005-logcollector-internal-1740] PASSED                                                                                                                                                                                     [ 80%]
tests/test_agent.py::test_get_config_error[002-logcollector-internal-1735] PASSED                                                                                                                                                                                     [ 82%]
tests/test_agent.py::test_get_config_error[000-None-None-1307] PASSED                                                                                                                                                                                                 [ 85%]
tests/test_agent.py::test_get_config_error[000-random-random-1101] PASSED                                                                                                                                                                                             [ 88%]
tests/test_agent.py::test_get_config_error[000-analysis-rules-1101] PASSED                                                                                                                                                                                            [ 91%]
tests/test_agent.py::test_get_config_error[000-analysis-internal-1013] PASSED                                                                                                                                                                                         [ 94%]
tests/test_agent.py::test_get_config_error[000-analysis-internal-1014] PASSED                                                                                                                                                                                         [ 97%]
tests/test_agent.py::test_get_config_error[000-analysis-internal-1101] PASSED                                                                                                                                                                                         [100%]

========================================================================================================================= 35 passed in 0.54 seconds =========================================================================================================================
```

Best regards,
Marta

### CODE REVIEW

File: framework/wazuh/configuration.py [hunk 1]
```diff
@@ -723,3 +704,62 @@ def upload_group_file(group_id, tmp_file, file_name='agent.conf'):
...
+
+
+def get_active_configuration(agent_id, component, configuration):
+    """
+    Reads agent loaded configuration in memory
+    """
+    if not component or not configuration:
+        raise WazuhException(1307)
+
+    sockets_path = common.ossec_path + "/queue/ossec/"
```
Comment (crd1985) [inline framework/wazuh/configuration.py]: Better use os.path.join :)

File: framework/wazuh/configuration.py [hunk 2]
```diff
@@ -723,3 +704,62 @@ def upload_group_file(group_id, tmp_file, file_name='agent.conf'):
...
+
+    components = ["agent", "agentless", "analysis", "auth", "com", "csyslog", "integrator", "logcollector", "mail",
```
Comment (crd1985) [inline framework/wazuh/configuration.py]: Searching in a list has a complexity O(n) while searching in a set has a complexity O(1).

File: framework/wazuh/configuration.py [hunk 3]
```diff
@@ -723,3 +704,62 @@ def upload_group_file(group_id, tmp_file, file_name='agent.conf'):
...
+        else:
+            raise WazuhException(1013, str(e))
+
+    # Send message
+    s.send(command.encode())
+
+    # Receive response
+    try:
+        # Receive data length
+        data = s.receive().decode().split(" ", 1)
```
Comment (crd1985) [inline framework/wazuh/configuration.py]: These lines are not covered by unit tests.

File: framework/wazuh/configuration.py (hunk 2 above)
Comment (mgmacias95) [reply framework/wazuh/configuration.py]: Fixed in https://github.com/wazuh/wazuh/pull/2993/commits/545fa149c9ff1c9dd31143dcaeca17d459d91861

File: framework/wazuh/configuration.py (hunk 1 above)
Comment (mgmacias95) [reply framework/wazuh/configuration.py]: Fixed in https://github.com/wazuh/wazuh/pull/2993/commits/545fa149c9ff1c9dd31143dcaeca17d459d91861

File: framework/wazuh/configuration.py (hunk 3 above)
Comment (mgmacias95) [reply framework/wazuh/configuration.py]: Done in 37432d86ad22e5a26ea27e7964a2b5523eb379e7

**Pull request merged**